src/rax.c

```diff
@@ -1673,6 +1673,7 @@ int raxSeek(raxIterator *it, const char *op, unsigned char *ele, size_t len) {
                  * node, but will be our match, representing the key "f".
                  *
                  * So in that case, we don't seek backward. */
+                it->data = raxGetData(it->node);
```

Comment:
nice catch.
I think we should sync from https://github.com/antirez/rax/blob/master/rax.c automatically to avoid this kind of bug.